### PR TITLE
feat/AB#86666_SUI-ABC-Proper-message-should-display-when-user-tries-to-access-SUI-application-without-permission

### PR DESCRIPTION
**Pull request by unai-reliefapp**
# Description

- fix: loading state flow in web components once application is correctly loaded, not access is correctly set, or an error is triggered on loading application 
- feat: add empty component to display error/access denied messages for web components dashboards. Use environment noAccessMessage set for web components in order to dynamically set current dashboard name inside the message. For that we check the current url hashed name using angular location 
- refactor: delete not used share component inside web components 
- fix: survey access when no survey is set if user has not access to dashboard(but detach logic is still triggered) 
- feat: avoid the use of snackbars for web components on dashboard loading

## Useful links

- [Link to ticket](https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/86666)

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] Improvement (refactor or addition to existing functionality)

# How Has This Been Tested?

Please refer to video below

## Screenshots

https://github.com/ReliefApplications/ems-frontend/assets/123092672/602179c3-ddf0-4be9-b186-400dbf600997

# Checklist:

( * == Mandatory ) 

- [X] * I have set myself as assignee of the pull request
- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have put the ticket for review, adding the oort-frontend team to the list of reviewers
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# More explanation
https://www.loom.com/share/05a716d61b9744faaf51fb304c21d1e5?sid=f87cf896-582a-4f76-93ae-8ceed801b145

### CODE REVIEW

File: apps/web-widgets/src/app/widgets/app-widget/application/pages/dashboard/dashboard.component.ts [hunk 1]
```diff
@@ -189,24 +195,23 @@ export class DashboardComponent
           this.contextService.isFilterEnabled.next(this.showFilter);
           this.contextService.setFilter(this.dashboard);
         } else {
+          // Get current url's hash to get the tab name and pass it to the no access message
+          let currentPageName =
```
Comment (AntoineRelief) [inline apps/web-widgets/src/app/widgets/app-widget/application/pages/dashboard/dashboard.component.ts:199]: @unai-reliefapp 
so, it's extracting the name from the url?

Comment (unai-reliefapp) [reply apps/web-widgets/src/app/widgets/app-widget/application/pages/dashboard/dashboard.component.ts:199]: Yep, as there is no application access, no page access, only place i can find it is from the hash in the url

Comment (unai-reliefapp) [reply apps/web-widgets/src/app/widgets/app-widget/application/pages/dashboard/dashboard.component.ts:199]: Default value if not found is Current page

Comment (AntoineRelief) [reply apps/web-widgets/src/app/widgets/app-widget/application/pages/dashboard/dashboard.component.ts:199]: okay, in that case we should prefer a generic message, it's really too specific to this use case @unai-reliefapp

Comment (unai-reliefapp) [reply apps/web-widgets/src/app/widgets/app-widget/application/pages/dashboard/dashboard.component.ts:199]: What does means to the code then?
Delete the env file updates or just add a default value in here?
If the hash is not found the message displayed uses "Current page"

Comment (AntoineRelief) [reply apps/web-widgets/src/app/widgets/app-widget/application/pages/dashboard/dashboard.component.ts:199]: I'll find a message

**Comment by AntoineRelief** [comment]
@unai-reliefapp 
deleting this one, I'll create another PR, reusing some of the logic, but improving it a bit as well
I'll send the link there when done

**Pull request closed without merging**